furnace.c

```diff
@@ -164,14 +186,133 @@ set_stirrer(bool opt)
 static void
 command_handler(furnace_context_t* ctx, uint8_t* buffer, void (*feedback)(const char*, const size_t))
 {
-  unsigned arg;
+  unsigned arg, arg2;
   char     str_arg[BUF_SIZE];
 
   if (buffer[0] == '\n') return;
 
   if (memcmp(buffer, "reboot", 6) == 0) {
     reset_usb_boot(0,0);
-  } else if (sscanf(buffer, "max_pwm %u", &arg) == 1) {
+  } 
+#if CONFIG_SPIN_COATER 
+#if CONFIG_SPIN_COATER == CONFIG_SPIN_COATER_PWM
+  else if (sscanf(buffer, "spin_pwm %u", &arg) == 1) {
+      const int ret = set_spin_coater_pwm_safe(ctx, arg);
+      if (ret) {
+        char msg[BUF_SIZE];
+        const size_t msg_len =
+          snprintf(msg,
+                  sizeof(msg),
+                  "pwm argument out of scope, should be [%d; %d]!\r\n",
+                  SPIN_COATER_PWM_490_FREQ_DUTY_MIN,
+                  SPIN_COATER_PWM_490_FREQ_DUTY_MAX);
+        feedback(msg, msg_len);
+        DEBUG_printf("Chaning PWM duty failed\n");
+        return;
+      }
+      if (arg == PWM_IDLE_DUTY) {
+        ctx->spin_coater.spin_state = SPIN_IDLE;
+        cancel_alarm(ctx->spin_coater.spin_timer);
+        const char msg[] = "spin stopped\r\n";
+        feedback(msg, sizeof(msg)-1);
+      } else {
+        ctx->spin_coater.spin_state = SPIN_STARTED_WITH_FORCE_VALUE;
+      }
+  }
+#elif CONFIG_SPIN_COATER == CONFIG_SPIN_COATER_DSHOT || CONFIG_SPIN_COATER == CONFIG_SPIN_COATER_DSHOT_WITH_TELEMETRY
```

Comment:
We tend to repeat the `CONFIG_SPIN_COATER == CONFIG_SPIN_COATER_DSHOT || CONFIG_SPIN_COATER == CONFIG_SPIN_COATER_DSHOT_WITH_TELEMETRY` often. Can we create some define alias like:
```cpp
#define CONFIG_SPIN_COATER_DSHOT_ANY (\
    CONFIG_SPIN_COATER == CONFIG_SPIN_COATER_DSHOT || \
    CONFIG_SPIN_COATER == CONFIG_SPIN_COATER_DSHOT_WITH_TELEMETRY \
)
```